src/middlewared/middlewared/plugins/iscsi.py

```diff
@@ -808,14 +808,17 @@ async def extent_serial(self, serial):
                                                    ['name', 'rnin', 'vnet'],
                                                    ['name', 'rnin', 'bridge']])
                        )[0]
-                mac = nic['state']['link_address'].replace(':', '')
+                mac = nic['state']['link_address'].replace(':', '').strip()
 
-                ltg = await self.query()
+                ltg = await self.query([], {'order_by': ['id']})
                 if len(ltg) > 0:
-                    lid = ltg[0]['id']
+                    lid = ltg[-1]['id']
                 else:
                     lid = 0
-                return f'{mac.strip()}{lid:02}'
+                if osc.IS_LINUX:
```

Comment:
hmm why mac differs for linux? maybe it shouldnt?

Reply:
To clarify, are you referring to the behaviour of us not setting 15 letter limit which we are doing for freebsd ?

Reply:
I guess.

Why is nic['state']['link_address'] the same between bsd and linux?

Reply:
Yes it is. We don't need to enforce the 15 letter limit in SCALE because this limit is specific to ctld.

Reply:
Perhaps the limit should be enforced in the ctl config generation then, and not here?

Reply:
I am not sure I follow, we should not allow the user to set a serial exceeding 15 chars ? So if we implicitly trim it in ctl config generation, do you think that's okay and we should proceed with it ?